classes/article/PublishedArticleDAO.inc.php

```diff
@@ -125,6 +128,81 @@ function getPublishedArticles($issueId) {
 	}
 
 	/**
+	 * Get all published article versions ids
+	 * @param $submissionId int
+	 * @param $contextId int optional
+	 * @param $order string optional
+	 * @return array
+	*/
+	function getPublishedSubmissionRevisionIds($submissionId, $contextId = null, $order = SORT_DIRECTION_DESC){
+		$params = array((int) $submissionId);
+		if ($contextId) {
+			$params[] = (int) $contextId;
+		}
+
+		$sql = 'SELECT DISTINCT ss.submission_revision
+			FROM 	submission_settings ss
+				LEFT JOIN published_submissions ps ON ps.submission_id = ss.submission_id
+				LEFT JOIN submissions s ON ps.submission_id = s.submission_id
+			WHERE 	ss.submission_id = ?'.
+				($contextId ? ' AND s.context_id = ?' : '').'
+				AND ss.setting_name = \'datePublished\'
+			ORDER BY ss.submission_revision ' . $this->getDirectionMapping($order);
+
+		$result = $this->retrieve($sql, $params);
+		$submissionRevisions = array();
+
+		foreach($result as $submission) {
+			$submissionRevisions[] = $submission['submission_revision'];
+		}
+
+		return $submissionRevisions;
+	}
+
+	/**
+	 * Get all published article versions
+	 * @param $submissionId int
+	 * @param $contextId int optional
+	 * @param $order string optional
+	 * @return array
+	*/
+	function getPublishedSubmissionRevisions($submissionId, $contextId = null, $order = SORT_DIRECTION_DESC){
+		$submissionDao = Application::getSubmissionDAO();
+		$publishedSubmissionRevisions = array();
+		$submissionRevisions = $this->getPublishedSubmissionRevisionIds($submissionId, $contextId, $order);
+
+		foreach($submissionRevisions as $revision){
+			$publishedSubmissionRevisions[] = $submissionDao->getById($submissionId, null, false, $revision);
+		}
+
+		return $publishedSubmissionRevisions;
+	}
+
+	/**
+	 * Get the latest published revision id for a submission
+	 * @param $submissionId int
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getLatestRevisionId($submissionId, $contextId = null){
+		$params = array((int) $submissionId);
+		if ($contextId) {
+			$params[] = (int) $contextId;
+		}
+
+		$sql = 'SELECT MAX(ss.submission_revision)
+			FROM 	submission_settings ss
+				INNER JOIN submissions s ON ss.submission_id = s.submission_id
+			WHERE 	ss.submission_id = ?
+				AND ss.setting_name = \'datePublished\'
```

Comment:
I don't think the setting_value can be empty, because the publication date can not be removed once it has been set (only changed to another date).